src/abcd_microstructure_pipelines/masks.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import itertools
+import logging
+import multiprocessing
+import os
+from collections.abc import Iterable
+from pathlib import Path
+
+import click
+import dipy.core.gradients
+import dipy.io
+import dipy.io.image
+
+logging.basicConfig(level=os.environ.get("LOG_LEVEL", "WARN"))
+
+
+def find_all_cases(root: Path) -> Iterable[Path]:
+    """
+    Recursively find all dwi cases in directory. Yield the case base names with file suffixes removed.
+    """
+    for dwi in root.rglob("*_dwi.nii.gz"):
+        yield dwi.with_name(dwi.name.removesuffix(".nii.gz"))
+
+
+def gen_b0_mean(dwi: Path, bval: Path, bvec: Path, b0_out: Path) -> None:
+    """
+    :param dwi: `_dwi.nii.gz` input
+    :param bval: `.bval` input
+    :param bvec: `.bvec` input
+    :param b0_out: `.b0.nii.gz` output
+    """
+    data, affine, img = dipy.io.image.load_nifti(str(dwi), return_img=True)
+    bvals, bvecs = dipy.io.read_bvals_bvecs(str(bval), str(bvec))
+
+    gtab = dipy.core.gradients.gradient_table(bvals, bvecs)
+    b0_mean = data[:, :, :, gtab.b0s_mask].mean(axis=3)
+
+    b0_out.parent.mkdir(parents=True, exist_ok=True)
+
+    logging.debug("generate %r", b0_out)
+    dipy.io.image.save_nifti(str(b0_out), b0_mean, affine, img.header)
+
+
+@click.command("gen_masks")
+@click.option("--inputs", "-i", required=True, type=Path)
+@click.option("--outputs", "-o", required=True, type=Path)
+@click.option("--overwrite", is_flag=True)
+@click.option("--parallel", "-j", is_flag=True)
+def gen_masks(inputs: Path, outputs: Path, overwrite: bool, parallel: bool) -> None:
+    """
```

Comment:
All in all I'm not sure if it's worth keeping the fields for the api docs, but I would prefer to keep them just for my IDE integration. I'll add `\f` so that help text is better.

Reply:
Thanks for explaining all that! Definitely it will be good to have nice docstrings everywhere: pycharm/vscode/whatever IDE, generated API doc html pages, and command line.